### PR TITLE
2197 cache rust

### PR DESCRIPTION
**Pull request by levBagryansky**
Closes: #2197
Rust caching. 
Saves `cargo project`'s target folder to `.eo/`. 
Copies it to local folder in the next compilation.

---

## PR-Codex overview
This PR focuses on updating the EO Maven Plugin by adding a new dependency and making changes to the BinarizeMojo class. 

### Detailed summary
- Added `commons-io` dependency to `pom.xml`
- Added imports and test methods to `BinarizeMojoTest.java`
- Updated `BinarizeMojo.java` with new `import` and changes to the `build` method

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by levBagryansky** [comment]
@mximp please review

**Comment by levBagryansky** [comment]
@Graur could you please review?

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java [hunk 1]
```diff
@@ -78,45 +83,82 @@ public final class BinarizeMojo extends SafeMojo {
...
+                    cached,
+                    target
+                )
+            );
+            FileUtils.copyDirectory(
+                cached,
+                target
+            );
+        }
+        final ProcessBuilder builder = new ProcessBuilder("cargo", "build")
```
Comment (Graur) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java:118]: @levBagryansky Could this variable be inlined?

File: eo-maven-plugin/src/test/java/org/eolang/maven/BinarizeMojoTest.java [hunk 2]
```diff
@@ -72,4 +75,25 @@ void failsWithIncorrectInsert(@TempDir final Path temp) throws IOException {
             () -> maven.execute(new FakeMaven.Binarize())
         );
     }
+
+    @Test
+    @Tag("slow")
```
Comment (Graur) [inline eo-maven-plugin/src/test/java/org/eolang/maven/BinarizeMojoTest.java:80]: @levBagryansky Could you please describe what is the purpose of this annotation ? E.g. without `@Suite` and `@IncludeTags` annotations?

**Review by Graur — CHANGES_REQUESTED** [review]
@levBagryansky Good one, just a few comments from my side:

File: eo-maven-plugin/src/test/java/org/eolang/maven/BinarizeMojoTest.java (hunk 2 above)
Comment (levBagryansky) [reply eo-maven-plugin/src/test/java/org/eolang/maven/BinarizeMojoTest.java:80]: @Graur this is from #2091. This test is executed in `Pqulice` only.

Comment (Graur) [reply eo-maven-plugin/src/test/java/org/eolang/maven/BinarizeMojoTest.java:80]: @levBagryansky I see. Thanks!

**Review by Graur — APPROVED** [review]
@levBagryansky Looks good to me
@yegor256 Could you please have a look

**Comment by levBagryansky** [comment]
@yegor256 could you please merge these changes?

**Comment by levBagryansky** [comment]
@yegor256 please check

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java [hunk 3]
```diff
@@ -78,45 +83,82 @@ public final class BinarizeMojo extends SafeMojo {
...
+                    target
+                )
+            );
+            FileUtils.copyDirectory(
+                cached,
+                target
+            );
+        }
+        Logger.info(this, "Building rust project..");
+        final Process building = new ProcessBuilder("cargo", "build")
```
Comment (yegor256) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java:119]: @levBagryansky you can use `VerboseProcess`, which we use in `MainTest`

**Comment by levBagryansky** [comment]
@Graur could you please check it again?

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java [hunk 4]
```diff
@@ -78,45 +77,73 @@ public final class BinarizeMojo extends SafeMojo {
...
+            );
+        }
+        Logger.info(this, "Building rust project..");
+        try (
+            VerboseProcess proc = new VerboseProcess(
+                new ProcessBuilder("cargo", "build")
+                    .directory(project)
+            )
+        ) {
+            proc.stdout();
```
Comment (Graur) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java:119]: @levBagryansky We can use `stdoutQuietly()` and avoid `IllegalArgumentException` if something was wrong with the build. But if that behaviour is intentional, that is ok.

Comment (levBagryansky) [reply eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java:119]: @Graur yes, this is intentional.

**Comment by levBagryansky** [comment]
@yegor256 Could you please take a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**